File: libs/power-log-parser/src/lib/controls/controls-manager.ts
import type { ActionParser } from '../action-parser';
import type { StateFacade } from '../state/state-facade';
import { StateType } from '../state/parser-state';
import { BG_EXCLUDED_PARSERS } from './battlegrounds-controls';

export class ControlsManager {
	private stateFacade: StateFacade;
	private stateType: StateType;

	constructor(stateFacade: StateFacade, stateType: StateType) {
		this.stateFacade = stateFacade;
		this.stateType = stateType;
	}

	Applies(parser: ActionParser): boolean {
		if (this.stateFacade.IsBattlegrounds()) {
			if (BG_EXCLUDED_PARSERS.has(parser.ParserName)) {
				return false;
			}
		}
		return true;
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Events/Parsers/Controls/ControlsManager.cs
using HearthstoneReplays.Parser;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HearthstoneReplays.Events.Parsers.Controls
{
    internal class ControlsManager
    {
        private StateFacade stateFacade;
        private StateType stateType;

        public ControlsManager(StateFacade stateFacade, StateType stateType)
        {
            this.stateFacade = stateFacade;
            this.stateType = stateType;
        }

        public bool Applies(ActionParser parser)
        {
            if (stateFacade.IsBattlegrounds())
            {
                if (BattlegroundsControls.EXCLUDED_PARSERS.Contains(parser.GetType())) {
                    return false;
                }
            }
            return true;
        }
    }
}
